sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/PathClientBuilder.java

```diff
@@ -0,0 +1,453 @@
+        this.storageSharedKeyCredential = Objects.requireNonNull(credential, "'credential' cannot be null.");
+        this.tokenCredential = null;
+        this.sasTokenCredential = null;
+        return this;
+    }
+
+    /**
+     * Sets the {@link TokenCredential} used to authorize requests sent to the service.
+     *
+     * @param credential The credential to use for authenticating request.
+     * @return the updated PathClientBuilder
+     * @throws NullPointerException If {@code credential} is {@code null}.
+     */
+    public PathClientBuilder credential(TokenCredential credential) {
+        blobClientBuilder.credential(credential);
+        this.tokenCredential = Objects.requireNonNull(credential, "'credential' cannot be null.");
+        this.storageSharedKeyCredential = null;
+        this.sasTokenCredential = null;
+        return this;
+    }
+
+    /**
+     * Sets the SAS token used to authorize requests sent to the service.
+     *
+     * @param sasToken The SAS token to use for authenticating requests.
+     * @return the updated PathClientBuilder
+     * @throws NullPointerException If {@code sasToken} is {@code null}.
+     */
+    public PathClientBuilder sasToken(String sasToken) {
+        blobClientBuilder.sasToken(sasToken);
+        this.sasTokenCredential = new SasTokenCredential(Objects.requireNonNull(sasToken,
+            "'sasToken' cannot be null."));
+        this.storageSharedKeyCredential = null;
+        this.tokenCredential = null;
+        return this;
+    }
+
+    /**
+     * Clears the credential used to authorize the request.
+     *
+     * <p>This is for blobs that are publicly accessible.</p>
+     *
+     * @return the updated PathClientBuilder
+     */
+    public PathClientBuilder setAnonymousAccess() {
+        blobClientBuilder.setAnonymousAccess();
+        this.storageSharedKeyCredential = null;
+        this.tokenCredential = null;
+        this.sasTokenCredential = null;
+        return this;
+    }
+
+    /**
+     * Sets the service endpoint, additionally parses it for information (SAS token, file system name, path name)
+     *
+     * <p>If the endpoint is to a file/directory in the root container, this method will fail as it will interpret the
+     * path name as the file system name. With only one path element, it is impossible to distinguish between a file
+     * system name and a path in the root file system, so it is assumed to be the file system name as this is much more
+     * common. When working with paths in the root file system, it is best to set the endpoint to the account url and
+     * specify the path name separately using the {@link PathClientBuilder#pathName(String) pathName} method.</p>
+     *
+     * @param endpoint URL of the service
+     * @return the updated PathClientBuilder object
+     * @throws IllegalArgumentException If {@code endpoint} is {@code null} or is a malformed URL.
+     */
+    public PathClientBuilder endpoint(String endpoint) {
```

Comment:
This should support IP style endpoints

Reply:
https://github.com/Azure/azure-sdk-for-java/issues/6118